gemrb/core/GUI/WindowManager.cpp

```diff
@@ -91,12 +92,14 @@ WindowManager::~WindowManager()
 
 void WindowManager::DestroyWindows(WindowList& list)
 {
-	WindowList::iterator it = list.begin();
-	for (; it != list.end(); ++it) {
-		Window* win = *it;
+	while (!list.empty()) {
```

Comment:
forgive me, I'm rusty still, but to me it looks like changing `Window::~Window` to invoke `manager.CloseWindow(this)` instead of `Close()` would not only fix this but prevent this issue from cropping up elsewhere.